### ISSUE TITLE
gocognit gocyclo support avg, could golangci-lint support it

### ISSUE THREAD

**Issue by daqingshu** [golangci/golangci-lint#887]
Thank you for creating the issue!

- [ X] Yes, I'm using a binary release within 2 latest major releases. Only such installations are supported.
- [ X] Yes, I've searched similar issues on GitHub and didn't find any.
- [ X] Yes, I've included all information below (version, config, etc).

Please include the following information:

<details><summary>Version of golangci-lint</summary>

```bash
$ golangci-lint --version
golangci-lint has version 1.21.0 built from 645e794 on 2019-10-15T18:16:56Z
```

</details>

origin gocognit support -avg argument

```bash
$ gocognit
Calculate cognitive complexities of Go functions.
Usage:
        gocognit [flags] <Go file or directory> ...
Flags:
        -over N   show functions with complexity > N only and
                  return exit code 1 if the set is non-empty
        -top N    show the top N most complex functions only
        -avg      show the average complexity over all functions,
                  not depending on whether -over or -top are set
The output fields for each line are:
<complexity> <package> <function> <file:row:column>
```

and gocyclo also support -avg argument

```bash
gocyclo -avg .
```

could golangci-lint support it?

**Comment by tpounds** [comment]
@daqingshu It looks like the `-avg` flag is intended to support reporting when using the tools directly. What is your intended use case here?

**Comment by daqingshu** [comment]
> @daqingshu It looks like the `-avg` flag is intended to support reporting when using the tools directly. What is your intended use case here?

I intended want use average complexity over all functions at merge gate. but I think it's not a good idea, I'll close this issue, thanks. 

who wrote complex function, who should refractory the function
